topology/probes/k8s/k8s.go

```diff
@@ -23,49 +23,87 @@
 package k8s
 
 import (
+	"github.com/skydive-project/skydive/config"
+	"github.com/skydive-project/skydive/logging"
 	"github.com/skydive-project/skydive/topology/graph"
 )
 
 const (
-	PodToContainerLink = "pod2container"
-	PolicyToPodLink    = "policy2pod"
+	podToContainerLink = "pod2container"
+	policyToPodLink    = "policy2pod"
 )
 
 var (
-	PodToContainerMetadata = graph.Metadata{"RelationType": PodToContainerLink}
-	PolicyToPodMetadata    = graph.Metadata{"RelationType": PolicyToPodLink}
+	podToContainerMetadata = graph.Metadata{"RelationType": podToContainerLink}
+	policyToPodMetadata    = graph.Metadata{"RelationType": policyToPodLink}
 )
 
-type probe struct {
+// Probe for tracking k8s events
+type Probe struct {
 	graph              *graph.Graph
 	client             *kubeClient
 	podCache           *podCache
 	networkPolicyCache *networkPolicyCache
+	nodeCache          *nodeCache
+	containerCache     *containerCache
 }
 
-func (k8s *probe) Start() {
-	k8s.networkPolicyCache.Start()
-	k8s.podCache.Start()
+type starter interface {
+	Start()
+	Stop()
 }
 
-func (k8s *probe) Stop() {
-	k8s.networkPolicyCache.Stop()
-	k8s.podCache.Stop()
+func (probe *Probe) getSubProbes() []starter {
```

Comment:
same here you could use the proble/bundle.go

Reply:
Replaced map of subProbes with probe.ProbeBundle